packages/wonder-blocks-modal/components/modal-launcher.test.js

```diff
@@ -59,4 +64,80 @@ describe("ModalLauncher", () => {
         wrapper.find("button").simulate("click");
         expect(wrapper.find("ModalLauncherPortal")).toHaveLength(1);
     });
+
+    test("Adds an `onClickCloseButton` prop to the provided `modal`", () => {
```

Comment:
These tests to have a lot of implementation details in them.  I think we should skew towards more black box testing in the sense that we shouldn't know the implementation details of the components only what their behavior should be.  In the case of the `onClickCloseButton`, what I'd actually like to test is that clicking the `close` button on a modal will dismiss the modal.  This will require us to actually mount the components, but I think the tests will be more straight forward both to read and write and they'll be more robust to changes in the implementation details over time.

Reply:
How much of a deal-breaker is that? IMO I think these are fine for the moment, albeit more brittle. Maybe we can go back later and try and improve them?

Reply:
Yeah, I'm not sure where to do the trade-off here. The shallow approach is fragile in the sense that it will need to be updated if we refactor the relationship between `ModalLauncher` and `ModalLauncherPortal`. But the mount-based approach is fragile in the sense that almost any breakage in `ModalLauncherPortal` would also cause this test to break.

Both are prone to failure, but I think the latter case would be surprising and confusing if it broke, whereas I think the former case would be rather unsurprising if it broke, and I don't expect the former breakage to actually happen more than once or twice in our component's lifetime. But if there's a third version of the test that would be chill in both cases, I'd strongly prefer that!

Reply:
Point taken about unrelated components causes breakages here.  I do think we should have some more full mount tests that verify basic things without worry about implementation details.  Since we already have unit tests we'll know which part of the code the breakage is in.  The benefit of the full mount tests it's easier to see what the behavior should when the component is actually in use.  They replace a certain level of manual testing.  I no longer have to click on the close button to verify that it does indeed close the modal.  It also provides a level of documentation about how all of the subcomponents are supposed to function together.  We can leave this for now, but we should definitely go back and add some of these tests before calling the modals done.

Reply:
In hindsight, I suppose this exact same class of problem is what led me to create `ModalBackdrop` and test it separately. Maybe I should move `cloneElement` into there? I'll try that.